plugins/scaffolder-backend/src/lib/files/serializeDirectoryContents.ts

```diff
@@ -44,19 +44,34 @@ export async function serializeDirectoryContents(
     dot: true,
     gitignore: options?.gitignore,
     followSymbolicLinks: false,
+    // In order to pick up 'broken' symlinks, we oxymoronically request files AND folders yet we filter out folders
+    // This is because broken symlinks aren't classed as files so we need to glob everything
+    onlyFiles: false,
     objectMode: true,
     stats: true,
   });
 
   const limiter = limiterFactory(10);
 
   return Promise.all(
-    paths.map(async ({ path, stats }) => ({
-      path,
-      content: await limiter(async () =>
-        fs.readFile(joinPath(sourcePath, path)),
-      ),
-      executable: isExecutable(stats?.mode),
-    })),
+    paths
+      .filter(({ dirent }) => !dirent.isDirectory())
+      .filter(({ dirent, path }) => {
+        if (!dirent.isSymbolicLink()) return true;
+        if (!fs.existsSync(resolveSafeChildPath(sourcePath, path))) return true; // We only want symlinks that DO NOT exist (yet)
```

Comment:
It's a little annoying, but I want to move these to promises using `fs-extra` or the native `{ promises as fs } from 'fs'`.

Reply:
Ok, this one I've had absolutely no luck with.

The suggested replacement path for `existsSync` (not deprecated just yet, unlike `fs.exists`) is to use `fs.access` or `fs.stat`. `stat` is used internally within `fs.existsSync` as `access` doesn't always work on Windows supposedly.

Making this filter block `async` or really just waiting for anything promise based appears to cause the pipeline to execute in parallel, which I suppose is what we want but it causes the `sym` files to pass through and end up being mapped. All of the promise variations cause this to happen so I've left this `fs.existsSync` for now because it's the only thing I can get to work reliably 😭